src/libcore/intrinsics.rs

```diff
@@ -565,8 +565,17 @@ extern "rust-intrinsic" {
     pub fn atomic_umax_rel<T>(dst: *mut T, src: T) -> T;
     pub fn atomic_umax_acqrel<T>(dst: *mut T, src: T) -> T;
     pub fn atomic_umax_relaxed<T>(dst: *mut T, src: T) -> T;
+
+    #[cfg(not(stage0))]
+    pub fn prefetch<T>(data: *const T, rw: i32, locality: i32, cache: i32);
```

Comment:
This is a terrible API, even for an intrinsic. What’s the rw? What’s the locality? What’s the cache? What are the meaning of values these arguments take?

Reading the LLVM’s docs it seems like `rw` could at least be a bool (I suppose LLVM uses i32 simply to allow themselves more expansion – not a problem for us because intriniscs are forever unstable). Similarly for `cache`, although it is a bit harder to draw the line.